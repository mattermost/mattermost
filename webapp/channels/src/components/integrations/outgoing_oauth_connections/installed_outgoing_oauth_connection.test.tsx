// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import type {OutgoingOAuthConnection} from '@mattermost/types/integrations';

import DeleteIntegrationLink from 'components/integrations/delete_integration_link';
import type {InstalledOutgoingOAuthConnectionProps} from 'components/integrations/outgoing_oauth_connections/installed_outgoing_oauth_connection';
import InstalledOutgoingOAuthConnection from 'components/integrations/outgoing_oauth_connections/installed_outgoing_oauth_connection';

describe('components/integrations/InstalledOutgoingOAuthConnection', () => {
    const team = {name: 'team_name'};
    const outgoingOAuthConnection: OutgoingOAuthConnection = {
        id: 'someid',
        create_at: 1501365458934,
        creator_id: 'someuserid',
        update_at: 1501365458934,
        audiences: ['https://myaudience.com'],
        client_id: 'someid',
        client_secret: '',
        grant_type: 'client_credentials',
        name: 'My OAuth Connection',
        oauth_token_url: 'https://tokenurl.com',
    };

    const baseProps: InstalledOutgoingOAuthConnectionProps = {
        team,
        outgoingOAuthConnection,
        creatorName: 'somename',
        onDelete: jest.fn(),
        filter: '',
    };

    test('should match snapshot', () => {
        const props = {...baseProps};
        const wrapper = shallow(
            <InstalledOutgoingOAuthConnection {...props}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should have called props.onDelete on handleDelete ', () => {
        const newOnDelete = jest.fn();
        const props = {...baseProps, team, onDelete: newOnDelete};
        const wrapper = shallow(
            <InstalledOutgoingOAuthConnection {...props}/>,
        );

        expect(wrapper.find(DeleteIntegrationLink).exists()).toBe(true);
        wrapper.find(DeleteIntegrationLink).props().onDelete();
        expect(newOnDelete).toHaveBeenCalled();
        expect(newOnDelete).toHaveBeenCalledWith(outgoingOAuthConnection);
    });
});
