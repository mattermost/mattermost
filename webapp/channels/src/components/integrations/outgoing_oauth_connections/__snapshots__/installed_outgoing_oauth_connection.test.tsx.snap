// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/integrations/InstalledOutgoingOAuthConnection should match snapshot 1`] = `
<div
  className="backstage-list__item"
>
  <div
    className="item-details"
  >
    <div
      className="item-details__row d-flex flex-column flex-md-row justify-content-between"
    >
      <strong
        className="item-details__name"
      >
        My OAuth Connection
      </strong>
      <div
        className="item-actions"
      >
        <Link
          to="/team_name/integrations/outgoing-oauth2-connections/edit?id=someid"
        >
          <MemoizedFormattedMessage
            defaultMessage="Edit"
            id="installed_integrations.edit"
          />
        </Link>
         - 
        <Connect(DeleteIntegrationLink)
          modalMessage={
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Deleting this connection will break any integrations using it"
              id="installed_outgoing_oauth_connections.delete.wanring"
            />
          }
          onDelete={[Function]}
          subtitleText={
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Are you sure you want to delete {connectionName}?"
              id="installed_outgoing_oauth_connections.delete.confirm"
              values={
                Object {
                  "connectionName": <strong>
                    My OAuth Connection
                  </strong>,
                }
              }
            />
          }
        />
      </div>
    </div>
    <div
      className="item-details__row"
    >
      <span
        className="item-details__token"
      >
        <MemoizedFormattedMessage
          defaultMessage="Client ID: "
          id="installed_integrations.client_id"
        />
        <strong>
          someid
        </strong>
      </span>
    </div>
    <div
      className="item-details__row"
    >
      <span
        className="item-details__token"
      >
        <MemoizedFormattedMessage
          defaultMessage="Client Secret: ********"
          id="installed_outgoing_oauth_connections.client_secret"
        />
      </span>
    </div>
    <div
      className="item-details__row"
    >
      <span
        className="item-details__url word-break--all"
      >
        <MemoizedFormattedMessage
          defaultMessage="Audience URLs: {urls}"
          id="installed_integrations.audience_urls"
          values={
            Object {
              "urls": "https://myaudience.com",
            }
          }
        />
      </span>
    </div>
    <div
      className="item-details__row"
    >
      <span
        className="item-details__url word-break--all"
      >
        <MemoizedFormattedMessage
          defaultMessage="Token URL: {url}"
          id="installed_integrations.token_url"
          values={
            Object {
              "url": "https://tokenurl.com",
            }
          }
        />
      </span>
    </div>
    <div
      className="item-details__row"
    >
      <span
        className="item-details__creation"
      >
        <MemoizedFormattedMessage
          defaultMessage="Created by {creator} on {createAt, date, full}"
          id="installed_integrations.creation"
          values={
            Object {
              "createAt": 1501365458934,
              "creator": "somename",
            }
          }
        />
      </span>
    </div>
  </div>
</div>
`;
